### PR TITLE
Revert #113588 to fix bootstrap timings

### PR DESCRIPTION
**Pull request by lqd**
This reverts #113588 which seems to have broken perf's bootstrap timings via some git issue

https://github.com/rust-lang/rust/pull/114318#issuecomment-1660807886 show a newly broken benchmark, the error at the time was 

```
fatal: Path 'src/ci/channel' exists on disk, but not in 'e62323df22ecf9c163023132d17b7114f68b72e8'.
       thread 'main' panicked at 'command did not execute successfully: cd "/home/collector/rustc-perf/rust" && "git" "show" "e62323df22ecf9c163023132d17b7114f68b72e8:src/ci/channel"
       expected success, got: exit status: 128', config.rs:1786:27
```

If this lands, it will reopen #101907 and annoy miri, but it could actually be an issue that would appear during the next bootstrap bump, not just rustc-perf today.

r? @ghost

### CODE REVIEW

**Comment by lqd** [comment]
Let's see if timings would be fixed
@bors try @rust-timer queue

File: src/tools/build_helper/src/git.rs [hunk 1]
```diff
@@ -78,22 +78,13 @@ pub fn rev_exists(rev: &str, git_dir: Option<&Path>) -> Result<bool, String> {
 /// We will then fall back to origin/master in the hope that at least this exists.
 pub fn updated_master_branch(git_dir: Option<&Path>) -> Result<String, String> {
     let upstream_remote = get_rust_lang_rust_remote(git_dir)?;
-    for upstream_master in [format!("{upstream_remote}/master"), format!("origin/master")] {
```
Comment (RalfJung) [inline src/tools/build_helper/src/git.rs:81]: Is it worth checking a revert of just the changes in updated_master_branch? Those are unrelated to the Miri changes, it just seemed reasonable to ensure the remove branch returned here actually exists.

**Comment by lqd** [comment]
there is bootstrap data on the server, so let's land this for now, and figure out how to land the initial PR after that

r? @RalfJung

**Comment by RalfJung** [comment]
I guess let's land this but I hope this doesn't mean https://github.com/rust-lang/rust/issues/101907 will never get fixed. :(

@bors r+ p=1

**Comment by lqd** [comment]
If it's just some transient .git state, we should be able to fix it on the server, fix the rustc-perf collector, and reland the PR to fix the issue.
